collector/nvidia.go

```diff
@@ -0,0 +1,305 @@
+// +build windows
+
+package collector
+
+/*
+#include "nvml.h"
+#include <windows.h>
+
+// definition of required import symbols
+nvmlReturn_t DECLDIR (*nvmlInit_v2Ptr)() = NULL;
+const DECLDIR char* (*nvmlErrorStringPtr)(nvmlReturn_t result) = NULL;
+nvmlReturn_t DECLDIR (*nvmlDeviceGetCount_v2Ptr)(unsigned int *deviceCount) = NULL;
+nvmlReturn_t DECLDIR (*nvmlDeviceGetHandleByIndex_v2Ptr)(unsigned int index, nvmlDevice_t *device);
+nvmlReturn_t DECLDIR (*nvmlDeviceGetNamePtr)(nvmlDevice_t device, char *name, unsigned int length);
+nvmlReturn_t DECLDIR (*nvmlDeviceGetGraphicsRunningProcesses_v2Ptr)(nvmlDevice_t device, unsigned int *infoCount, nvmlProcessInfo_t *infos);
+nvmlReturn_t DECLDIR (*nvmlSystemGetProcessNamePtr)(unsigned int pid, char *name, unsigned int length);
+nvmlReturn_t DECLDIR (*nvmlDeviceGetDriverModelPtr)(nvmlDevice_t device, nvmlDriverModel_t *current, nvmlDriverModel_t *pending);
+nvmlReturn_t DECLDIR (*nvmlDeviceGetMemoryInfoPtr)(nvmlDevice_t device, nvmlMemory_t *memory);
+
+nvmlReturn_t invoke_nvmlInit_v2()
+{
+	return (*nvmlInit_v2Ptr)();
+}
+
+const char *invoke_nvmlErrorString(nvmlReturn_t error)
+{
+	return (*nvmlErrorStringPtr)(error);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetCount_v2(unsigned int *deviceCount)
+{
+	return (*nvmlDeviceGetCount_v2Ptr)(deviceCount);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetHandleByIndex_v2(unsigned int index, nvmlDevice_t *device)
+{
+	return (*nvmlDeviceGetHandleByIndex_v2Ptr)(index, device);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetName(nvmlDevice_t device, char *name, unsigned int length)
+{
+	return (*nvmlDeviceGetNamePtr)(device, name, length);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetGraphicsRunningProcesses_v2(nvmlDevice_t device, unsigned int *infoCount, nvmlProcessInfo_t *infos)
+{
+	return (*nvmlDeviceGetGraphicsRunningProcesses_v2Ptr)(device, infoCount, infos);
+}
+
+nvmlReturn_t invoke_nvmlSystemGetProcessName(unsigned int pid, char *name, unsigned int length)
+{
+	return (*nvmlSystemGetProcessNamePtr)(pid, name, length);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetDriverModel(nvmlDevice_t device, nvmlDriverModel_t *current, nvmlDriverModel_t *pending)
+{
+	return (*nvmlDeviceGetDriverModelPtr)(device, current, pending);
+}
+
+nvmlReturn_t invoke_nvmlDeviceGetMemoryInfo(nvmlDevice_t device, nvmlMemory_t *memory)
+{
+	return (*nvmlDeviceGetMemoryInfoPtr)(device, memory);
+}
+*/
+import "C"
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func init() {
+	registerCollector("nvidia", newNvidiaCollector, "NVIDIA")
```

Comment:
You don't seem to unmarshal the `NVIDIA` counters you request here?

Reply:
This collector does not take anything from perflib, all the WDDM stuff (GPU usage from perflib) has been put in the process collector.

Reply:
Sorry, this is literally just an error, since I don't think I initially understood what that parameter does, will remove it.